### DISCUSSION TITLE
Gameboy Checksums Wrong?

### DISCUSSION THREAD

**Discussion by stewartspicer** [sanni/cartreader#398] [Need help]
Hey all, hopefully a simple question. My gameboy games seem to be getting the wrong checksums, consistently. For example, the checksum for Pokemon Mystery Dungeon - Red Team Rescue (USA) on my reader came out to 99, for some reason. I'm not sure why it's 2 digit in the first place. It DOES read the cartridge name and ID successfully, so I'm wondering if this is a software issue?

Hardware 4, clock generator installed and enabled

**Comment by PsyK0p4T** [comment]
Do you have this for every games tested ? Did it work fine before ?
You can try to clean the cart pins ; change your sd with another one ; check if the game pcb are a real retail one and not a repro...

**Reply by stewartspicer** [reply]
Yeah, it's every cart, but I just built it. I'm still suspicious that something is wrong because the CRC is only 2 letters long for every game I've tried

**Comment by sanni** [comment]
2 letter checksum is normal for GBA, it's just the internal rom checksum not the CRC32. GBA carts need to be super clean to dump correctly right now, the current code still needs some tweaks like enabling the pull-ups. 
Until then take a q-tip and dip it in rubbing alcohol then go over the gold contacts of the GBA cartridge's edge connector until the q-tip does not get dark anymore.

**Reply by stewartspicer** [reply]
Hm, still no luck. It DOES pull the cart ID and version, but can't seem to get the rom and save size info. Works perfectly fine if I manually enter that data, though

My GBA library is one of my smallest retro libraries, so I don't mind putting that extra effort in. Thanks for the advice on this

**Reply by sanni** [reply]
Are you on a Mac or on Linux maybe? The databases rely on the Windows style line endings and maybe Mac/Linux mixes something up. 
Also I did change how the database works recently, maybe you got an older database version on the SD card that does not match the Cart_Reader Arduino sketch you have uploaded.

**Reply by stewartspicer** [reply]
Nope, I'm on windows. Just double checked, and I do have the windows-style \r\n line endings.

Database is the same as the current version on github, with a title line, an info line, and then a blank line.

So... Not really sure what it could be. I've checked the files in an emulator, and they seem to be fine. I have to manually enter the ROM and save info, but it seems to work when I do

**Comment by sanni** [comment]
Try flashing the 8.4 Release to the Arduino and copy the database files from that release to the SD. https://github.com/sanni/cartreader/releases/tag/V8.4

Maybe there is an error in the new database/new Arduino code that only affects certain carts.

**Reply by stewartspicer** [reply]
Short version: It was my fault, but I have it working now.

Long version: I was using the GBA.txt from the github, not the one in the release folder. Since I was on 8.4 and it expects there to be no game names, it wasn't matching. 

Suggestion for the next release: Add a "break;" between lines 830 and 831 in GBA.ino, in getCartInfo_GBA(). The way it is now, it checks the entire gba.txt, even if it matches on the first line
